### ISSUE TITLE
[Bug]: type errors with latest @remix-run/router

### ISSUE THREAD

**Issue by tchak** [remix-run/react-router#8950]
### What version of React Router are you using?

@remix-run/router 0.2.0-pre.0

### Steps to Reproduce

Run `tsc` in a project importing `@remix-run/router`

* `skipLibCheck` is `true`
* the project is not using `React`

### Expected Behavior

No type errors

### Actual Behavior

```
node_modules/@remix-run/router/history.ts:429:46 - error TS2304: Cannot find name '__DEV__'.

429 const readOnly: <T>(obj: T) => Readonly<T> = __DEV__
                                                 ~~~~~~~

node_modules/@remix-run/router/router.ts:853:11 - error TS2304: Cannot find name '__DEV__'.

853       if (__DEV__) {
              ~~~~~~~

node_modules/@remix-run/router/utils.ts:72:13 - error TS2503: Cannot find namespace 'React'.

72   element?: React.ReactNode;
               ~~~~~

node_modules/@remix-run/router/utils.ts:78:18 - error TS2503: Cannot find namespace 'React'.

78   errorElement?: React.ReactNode;
                    ~~~~~

Found 4 errors in 3 files.
```

I am also getting a run time error about `__DEV__`

**Comment by brophdawg11** [comment]
This looks to be a bug in the [changesets](https://github.com/changesets/changesets) workflow we were testing out for the most recent prerelease, so for now I would ignore the new packages and stick with `6.4.0-pre.2` and `@remix-run/router@0.1.0`.  We'll get updated versions out soon once we resolve the changesets issues.

**Comment by Brian-McBride** [comment]
Any workarounds with this?
I'm using some of the features to get a new lib for Ionic routing, but it seems 6.4.0 is really where it's at. 

Are you all just defining `__DEV__` as a global type?

**Comment by tchak** [comment]
This is fixed in [0.2.0-pre.1](https://www.npmjs.com/package/@remix-run/router/v/0.2.0-pre.1)
